notebooks/10_bayesian_networks_sampling/index.md

```diff
@@ -0,0 +1,233 @@
+
+Sampling has two main advantages:
+
+- Learning: By getting samples from an unknown distribution, we can learn the associated probabilities.
+- Performance: Getting a sample is much faster than  computing the right answer.
+
+The primitive element in any sampling algorithm is the generation of samples from a known probability distribution. So the step-by-step algorithm is described in the following section.
+
+# Sampling from Given Distribution
+Consider the example of picking random colored cubes from a large bag of them. The probrability distribution of the colors of these cubes is given in the table below. In this setting, sampling is analogous to picking a random cube from said bag so that the probabilities are taken into account.
+<center>
+
+|   C   	| Pr(C) 	|
+|:-----:	|:----:	|
+|  green  	|  0.6 	|
+|   red 	|  0.1 	|
+|  blue 	|  0.3 	|
+
+</center>
+
+In order to draw a sample from this distribution, we can use a uniform distribution to generate a seed 
+and determine the sample based on that. As shown in the figure below, the unit length segment is 
+partitioned into parts named $l_i$, each having length equal to $p_i$. These $p_i$s represent the 
+probabilities of the distribution, and are equal to $Pr(C=c_i)$. In other words, we have indexed the 
+values that the random variable $C$ can take and defined the probabilities based on them. Here, we have 
+$c_1 = green$, $c_2 = red$ and $c_3 = blue$.
+
+![Partition of the Unit Length Segment](Images/Unit_Length_Segment_Partition.png "Partition of the Unit Length Segment")
+
+We say that the sample drawn has the value $c_i$, if the seed chosen lies in $l_i$. Since we chose the seed from a uniform distribution, it can be shown that the probability of the sample being equal to $c_i$ is $Pr(C=c_i)$.
+
+Thus, using uniform distributions in this manner, we can generate samples from a given distribution having the same corresponding probabilities.
+
+# Prior Sampling
+Now we're ready to tackle the sampling problem! Suppose we want to sample from the joint distribution 
+of the Bayes' Net, $Pr(x_1,...,x_n)$.
+
+Consider the following process:
+
+1.&nbsp; For i = 1 to n
+
+2.&emsp;&emsp; Sample $x_i$ from $Pr(x_i | parents(x_i))$
+
+3.&nbsp; Return $(x_1,...,x_n)$
+
+This process generates a sample $X=(x_1,...,x_n)$ with the following probability:
+$$
+S_{PS}(x_1,...,x_n) = \prod_{i=1}^{n}Pr(x_i|parents(x_i)) = Pr(x_1,...,x_n),
+$$
+which is the Bayes' Net's joint probability.
+
+Now we need to prove that the samples are realizations of I.I.D (Independent, Identically Distributed) random variables. 
+1. Since we choose a new random number for every sample, thus samples are independent from each other. 
+2. It has been also shown that they are identically distributed because of the formula above.
+
+Suppose we have drawn $N$ samples from the joint distribution. Let the number of samples of an event be 
+$N_{PS}(x_1,...,x_n)$.
+
+So based on the law of large numbers, we have:
```

Comment:
Readers may not know what the "law of large numbers" is, so it would be great if you can have hyperlinks here

Reply:
Thank you for your notice. It has been resolved.